compiler/src/iree/compiler/Dialect/Util/Transforms/PassDetail.h

```diff
@@ -16,6 +16,19 @@ namespace iree_compiler {
 namespace IREE {
 namespace Util {
 
+// Defines the names (compile phases) that can be used with TraceFrameMarkPass.
```

Comment:
since the tracing passes are specific to tracing utils, we should just move them into there - we don't need to use tablegen for them and can just have C++ classes in TracingUtils.cpp - we want to keep this as contained as we can

Reply:
> we don't need to use tablegen for them and can just have C++ classes in TracingUtils.cpp

Most (all?) of our passes are defined in tablegen. Can you point me at a sample pass written in just C++? (Fine if if you don't have one on hand but would save me some searching)